pkg/workos_errors/http.go

```diff
@@ -43,6 +43,34 @@ func isJsonResponse(r *http.Response) bool {
 	return strings.Contains(r.Header.Get("Content-Type"), "application/json")
 }
 
+// ErrorWithMessage struct to hold a single error with a message
+type ErrorWithMessage struct {
+	Message string `json:"message"`
+}
+
+// ResponseErrors is a custom type that can handle unmarshaling []string or []ErrorWithMessage
+type ResponseErrors []string
+
+func (e *ResponseErrors) UnmarshalJSON(data []byte) error {
+	// Try unmarshaling as []string
+	var stringErrors []string
+	if err := json.Unmarshal(data, &stringErrors); err == nil {
+		*e = stringErrors
+		return nil
+	}
+
+	// Try unmarshaling as []ErrorWithMessage
+	var structErrors []ErrorWithMessage
+	if err := json.Unmarshal(data, &structErrors); err == nil {
+		for _, se := range structErrors {
+			*e = append(*e, se.Message)
+		}
+		return nil
+	}
+
+	return fmt.Errorf("Errors field is not a valid format")
```

Comment:
Since we don't need to format anything in the error, we can `return errors.New("Errors field is not a valid format")` here instead.